### DISCUSSION TITLE
C grammar example

### DISCUSSION THREAD

**Discussion by fknfilewalker** [taocpp/PEGTL#358] [Q&A] [answered]
Is there an example that shows how to parse a c grammar? Or is this too much for PEGTL?

**Comment by ColinH** [comment, marked as answer]
We do not currently have an example for the C grammar. I don't think it would be too much for the PEGTL to implement a C parser, however it's a bit too much for an example that we would write just "for fun", i.e. without having a serious use case for it. We'd be happy to give support if you were to write one, and are always open to contributions ;-)

**Reply by fknfilewalker** [reply]
Sounds great.

For reference I mostly go with these grammars.
https://github.com/pointlander/peg/blob/master/grammars/c/c.peg 
https://learn.microsoft.com/en-us/cpp/c-language/lexical-grammar

One problem I am currently running into is, how to structure everything in a way that allows me to use the 'must' rule for parsing errors. Oftentimes you need to look far ahead to know if you really match the current rule, but then must already fires if it is the wrong one. Do you have any good solutions to that?

**Reply by ColinH** [reply]
Can you show an example of this difficulty?
